tiledb/sm/fragment/test/unit_fragment_identifier.cc

```diff
@@ -268,3 +268,11 @@ TEST_CASE("FragmentID: Valid uris", "[fragment_id][valid_uri]") {
     }
   }
 }
+
+TEST_CASE("FragmentID: has_fragment_name", "[fragment_id][has_fragment_name]") {
```

Comment:
FWIW this is an excellent opportunity to try out using rapidcheck since the properties and/or generators should be pretty straightforward.  Two that come to mind:
- `has_fragment_name` returns true on all valid fragment IDs
- the `FragmentID` constructor throws if and only if `has_fragment_name` returns false

Reply:
How about a 3 bullet guide/README sort of thingy, on how one can add a rapidcheck test in Core? 🙏

Reply:
+1 on the above, this sounds like it'd come in handy